src/datadoc/backend/model_validation.py

```diff
@@ -131,17 +140,24 @@ def check_obligatory_variables_metadata(self) -> Self:
             ObligatoryVariableWarning: If not all obligatory variable metadata fields
                 are filled in.
         """
-        if (
-            self.variables is not None
-            and num_obligatory_variables_fields_completed(
-                self.variables,
-            )
-            != NUM_OBLIGATORY_VARIABLES_FIELDS
-        ):
-            warnings.warn(
-                f"{OBLIGATORY_METADATA_WARNING} {get_missing_obligatory_variables_fields(self.variables)}",
+        if self.variables is not None:
+            for v in self.variables:
+                if (
+                    num_obligatory_variable_fields_completed(
+                        v,
+                    )
+                    != OBLIGATORY_VARIABLES_METADATA_IDENTIFIERS
```

Comment:
It looks like we're comparing an `int` to a `list` here?
```suggestion
                    != len(OBLIGATORY_VARIABLES_METADATA_IDENTIFIERS)
```